api/core/workflow/workflow_engine_manager.py

```diff
@@ -30,6 +31,8 @@
     WorkflowNodeExecutionStatus,
 )
 
+WORKFLOW_MAX_EXECUTION_TIME = int(os.environ.get('WORKFLOW_MAX_EXECUTION_TIME', str(600)))  # 600s
```

Comment:
Suggest putting this variable in config.py and accessing it through current_app.config.get().